Framework/API/src/ExperimentInfo.cpp

```diff
@@ -889,12 +912,13 @@ ExperimentInfo::getInstrumentFilename(const std::string &instrumentName,
  * this reference.
  */
 const DetectorInfo &ExperimentInfo::detectorInfo() const {
-  if (!m_detectorInfo) {
+  if (!m_detectorInfoWrapper) {
     std::lock_guard<std::mutex> lock{m_detectorInfoMutex};
-    if (!m_detectorInfo)
-      m_detectorInfo = Kernel::make_unique<DetectorInfo>(getInstrument());
+    if (!m_detectorInfoWrapper)
+      m_detectorInfoWrapper =
+          Kernel::make_unique<DetectorInfo>(getInstrument());
   }
-  return *m_detectorInfo;
+  return *m_detectorInfoWrapper;
```

Comment:
Ok. I'll have to read a bit more to establish the interplay with all the existing types.